### ISSUE TITLE
Rewrite User system to support SQL structure.

### ISSUE THREAD

**Issue by fabianmakila** [fabianmakila/ProxyChat#48]
This will be a complete rewrite of the current user system. While I've been fairly happy with the file based storage system this needs to eventually happen in order to support future features.

I've been wanting to support MySQL storage for a while now but haven't had a really good reason to do a complete rewrite.

The biggest motivation for this is that this makes OfflinePlayer type argument much easier to implement. The current implementation that the mute / unmute uses is a bit wonky. Moving to a SQL based storage allows us to query x amount of users based of the user input which will be much more efficient than looking up the name from a JSON file.

TLDR;
- Move away from the "User" terminology and use "ProxyChatPlayer" instead.
- Switch to a SQL based storage system
- Support both SQLite and MySQL (maybe more in the future?)
